android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -42,6 +42,43 @@ import mozilla.components.support.ktx.kotlin.isSameOriginAs
 import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
 import mozilla.components.support.utils.Browsers
 
+/**
+ * The name of the file to be downloaded.
+ */
+@JvmInline
+value class Filename(val value: String)
```

Comment:
Tried using this [new inline class feature](https://kotlinlang.org/docs/inline-classes.html) as a way to help clients better understand what each parameter of the new callbacks represent.
Without this clients would see a list of parameter types eg: String, Int, Int without no useful context.
The inline classes are compiled to the underlying types so they incurs no performance penalty - they can be seen as type aliases and help for clients to easier infer what the lambda parameters represent and easily avoid bugs / ensure correct usages.